### ISSUE TITLE
Tray icon does not show in the tray area

### ISSUE THREAD

**Issue by AntonioFasano** [keybase/keybase-issues#3839]
In _Arch Linux and derivatives_, at system startup, **Keybase version 5.2.0** does not show its icon in the tray area, therefore there is no way to open the GUI in order to use it. 

Launching from the menu or running again `keybase-gui` from the terminal does not show the GUI nor the tray icon too.

Running from the terminal:

    electron /usr/share/keybase-app

shows the GUI, but not the tray icon and I also get the warning:

    Only one instance of keybase GUI allowed, bailing!

To avoid the warning I can use:

    systemctl stop --user keybase.gui && electron /usr/share/keybase-app&

but this does not help showing the tray icon.

See also Arch forum [thread](https://bbs.archlinux.org/viewtopic.php?id=252958)

**Comment by heronhaye** [comment]
Yes, not having the systray icon is a known "feature" of Electron 8. But that shouldn't prevent you from opening the app. What DE/WM are you using?

cc @eli-schwartz, who maintains the Arch community packages.

**Comment by eli-schwartz** [comment]
https://bugs.archlinux.org/task/65483

electron 8 doesn't show tray icons via XEmbed anymore, you need StatusNotifier/AppIndicator support in your DE. And if it starts backgrounded, then it is difficult to raise it to the foreground. ;)

The current keybase-gui script tries to start keybase.gui.service similar to how the run_keybase script in the keybase-bin package handles it. But that will merely check to see that it is started, it won't re-foreground the application.

Given the likelihood of people running desktop environments without StatusNotifier support and then being unable to interact with Keybase at all, I'm pondering to no longer use the keybase.gui.service in the launch path and just making that script a wrapper over `electron /usr/share/keybase-app`. This would always re-foreground.

@heronhaye what do you think?

**Comment by AntonioFasano** [comment]
@heronhaye 
I am using [awesome](https://awesomewm.org)

**Comment by marceloneil** [comment]
@eli-schwartz what do you think of:
```
systemctl --user is-active --quiet keybase.gui && electron /usr/share/keybase-app || systemctl --user start keybase.gui
```

**Comment by eli-schwartz** [comment]
@marceloneil I'm going to test out this solution in combination with v5.3.0 and hopefully push packages later tonight. Thanks for the suggestion.

**Comment by eli-schwartz** [comment]
I'm pushing the packages as I write this. You'll be especially interested in this change:

https://git.archlinux.org/svntogit/community.git/commit/trunk?h=packages/keybase&id=7c3e4552a783343f05df7cb0f913148abc24e0a6

Hopefully this will be enough to stem much of the confusion. I'll leave @heronhaye to figure out and/or answer the question of whether Keybase should more gracefully handle the systray feature. While I don't know that they will reimplement the XEmbed tray specification on their own to work around Electron no longer providing one, we chatted a bit in private and one alternative possibility might be detecting that there is no tray icon, and refusing to background. (I'd prefer it to just minimize in that case, wouldn't you? :wink:)

**Comment by AntonioFasano** [comment]
The problem is just message notifications, when you miss the pop-up. Probably one could add a separate notification app, not tied to Electron. I mean something very simple, just an icon that appears to signal you have unread messages, then you open keybase and verify.

**Comment by kwendenarmo** [comment]
+1 the tray icon does not show on linux mint 19.3 with keybase 5.3.0-20200310205642+4f2689009b

**Comment by scaprile** [comment]
Hi, I'm having the same issue on CentOS 7 since updated to 5.3.0.20200310205642.4f2689009b
Same reason ? Do I need to open a separate issue ?
Can I run some tests for you guys ?
Thx

**Comment by dietmarw** [comment]
I can confirm this too on Ubuntu 18.04 Mate flavour.

**Comment by scaprile** [comment]
> Same reason ? (I bet yes)

Yes, I'm right

> Do I need to open a separate issue ? (I bet no, all I need is "something" (stable) to provide StatusNotifierItem/AppIndicator on Gnome. The "stable" part is the most difficult, some shell extensions are prone to crashes)

Yes, I need something like this extension: [https://extensions.gnome.org/extension/615/appindicator-support/](https://extensions.gnome.org/extension/615/appindicator-support/)
Icons are really small and activation requires a non-intuitive double click, but does the job now.

@dietmarw , @kwendenarmo , and maybe others using Gnome and looking out for a solution, this might help you.
Cheers

**Comment by dietmarw** [comment]
Won't probably not work for me since the Mate desktop does not use Gnome-Shell.

**Comment by casasnovas** [comment]
Don't have the systray nor notifications on v5.3.1/Gentoo/awesomewm either.

**Comment by AntonioFasano** [comment]
At this point the only possibility workaround seems to set a cron job from the Keybase CLI
Do you know how to obtain unread messages from `keybase chat read` or the likes?

**Comment by edmundlaugasson** [comment]
To be honest then even after regular DEB-package installation there is no systray icon (_keybase version 5.5.1-20200527202541+39ca0071e5_). Using official Keybase repository in Linux Mint MATE desktop - _deb http://prerelease.keybase.io/deb stable main_ as guided here https://keybase.io/docs/the_app/install_linux

**Comment by eli-schwartz** [comment]
Hmm.

I mentioned this to @heronhaye on June 8. Electron upstream restored the XEmbed systray in https://github.com/electron/electron/pull/23674 and backported it to every major release from 8-10. Updating electron is sufficient to fix this...

It is in the release notes for 8.3.2 and 9.0.3 stable. keybase still uses 8.0.2.

If you use Arch Linux, the official [community] package uses 9.1.2 with the fix.

**Comment by AntonioFasano** [comment]
Problem solved for me. My current Keybase version is 5.5.1

**Comment by dietmarw** [comment]
The issue still exist in version 5.5.1 on Ubuntu 20.04. Therefore I think it is a good idea to keep this open until keybase has be repackages with the fixed electron version.

**Comment by elucidsoft** [comment]
Same issue on XFCE

**Comment by dietmarw** [comment]
I just got an update on my Ubuntu 20.04 machine and now with keybase version 5.5.2-20201016183637+d4ebf7d999
the tray icon is showing again.

**Comment by AntonioFasano** [comment]
Waiting for @elucidsoft confirmation to close.

**Comment by MrPrimate** [comment]
Resolved for me on Ubuntu 20.04 under i3. 🙇🏻
